geode-membership/src/main/java/org/apache/geode/distributed/internal/membership/gms/GMSMembership.java

```diff
@@ -1360,13 +1319,8 @@ public boolean verifyMember(ID mbr, String reason) {
 
   @Override
   public ID[] getAllMembers(final ID[] arrayType) {
-    latestViewReadLock.lock();
-    try {
-      List<ID> keySet = latestView.getMembers();
-      return keySet.toArray(arrayType);
-    } finally {
-      latestViewReadLock.unlock();
-    }
+    final List<ID> keySet = latestView.getMembers();
```

Comment:
Is the latest view and/or members list immutable?

Reply:
`latestView` isa `MembershipView` and yes, this PR makes `MembershipView` immutable (see `MembershipView.java`)

Reply:
This PR also makes `members` list unmodifiable.

Reply:
So if the view is immutable, what is the r/w lock protecting anymore?

Reply:
The lock is protecting new view formation in processView().  This data needs to be stable while the new view is created.